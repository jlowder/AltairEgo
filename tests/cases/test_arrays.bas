10 REM ARRAY AND STRING TESTS
20 PRINT "=== ARRAY TESTS ==="
30 DIM A(10), B(5,3), C$(5)
40 REM Test 1D array
50 FOR I=0 TO 10
60 A(I) = I * I
70 NEXT I
80 FOR I=0 TO 10
90 PRINT "A("; I; ")="; A(I)
100 NEXT I
110 REM Test 2D array
120 FOR I=0 TO 5
130 FOR J=0 TO 3
140 B(I,J) = I + J
150 NEXT J
160 NEXT I
170 FOR I=0 TO 5
180 FOR J=0 TO 3
190 PRINT "B("; I; ","; J; ")="; B(I,J)
200 NEXT J
210 NEXT I
220 REM Test string array
230 C$(0) = "ZERO"
240 C$(1) = "ONE"
250 C$(2) = "TWO"
260 FOR I=0 TO 2
270 PRINT "C$("; I; ")="; C$(I)
280 NEXT I
290 PRINT "=== STRING MANIPULATION ==="
300 A$ = "HELLO WORLD"
310 PRINT "Original: "; A$
320 PRINT "Length: "; LEN(A$)
330 PRINT "Left 5: "; LEFT$(A$, 5)
340 PRINT "Right 5: "; RIGHT$(A$, 5)
350 PRINT "Mid 3,5: "; MID$(A$, 3, 5)
360 REM Test string comparison
370 IF "ABC" < "DEF" THEN PRINT "ABC < DEF"
380 IF "HELLO" = "HELLO" THEN PRINT "Strings equal"
390 END
