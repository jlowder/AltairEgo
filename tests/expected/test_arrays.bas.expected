Altair Ego: Emulating Altair BASIC 32K Rev. 3.2
OK
OK
=== ARRAY TESTS ===
A( 0 )= 0
A( 1 )= 1
A( 2 )= 4
A( 3 )= 9
A( 4 )= 16
A( 5 )= 25
A( 6 )= 36
A( 7 )= 49
A( 8 )= 64
A( 9 )= 81
A( 10 )= 100
B( 0           0 )= 0
B( 0           1 )= 1
B( 0           2 )= 2
B( 0           3 )= 3
B( 1           0 )= 1
B( 1           1 )= 2
B( 1           2 )= 3
B( 1           3 )= 4
B( 2           0 )= 2
B( 2           1 )= 3
B( 2           2 )= 4
B( 2           3 )= 5
B( 3           0 )= 3
B( 3           1 )= 4
B( 3           2 )= 5
B( 3           3 )= 6
B( 4           0 )= 4
B( 4           1 )= 5
B( 4           2 )= 6
B( 4           3 )= 7
B( 5           0 )= 5
B( 5           1 )= 6
B( 5           2 )= 7
B( 5           3 )= 8
C$( 0 )=ZERO
C$( 1 )=ONE
C$( 2 )=TWO
=== STRING MANIPULATION ===
Original: HELLO WORLD
Length:  11
Left 5: HELLO
Right 5: WORLD
Mid 3,5: LLO W
ABC < DEF
Strings equal
OK
